File: packages/schemas/src/network.ts
// ── Network entities ──────────────────────────────────────────────────────────

export type AircraftType = "B737-800" | "A320" | "E175" | "B757-200"

export interface Airport {
  icao: string
  name: string
  city: string
  lat: number
  lon: number
  hourly_capacity: number
  elevation_ft: number
}

export interface Aircraft {
  tail_number: string
  aircraft_type: AircraftType
  seat_capacity: number
  base_airport: string
  min_turn_minutes: number
  current_airport?: string
}

export interface ScheduledFlight {
  id: string
  flight_number: string
  origin: string
  destination: string
  scheduled_departure: string   // ISO-8601
  scheduled_arrival: string     // ISO-8601
  tail_number: string
  aircraft_type: AircraftType
  passengers: number
  status?: FlightStatus
}

export type FlightStatus =
  | "scheduled"
  | "delayed"
  | "cancelled"
  | "diverted"
  | "airborne"
  | "landed"

export interface CrewPairing {
  id: string
  captain_id: string
  first_officer_id: string
  base: string
  flights: string[]            // ordered flight IDs
  report_time: string          // ISO-8601
  release_time: string         // ISO-8601
  duty_minutes: number
  flight_time_minutes: number
}

File: packages/schemas/python/network.py
from __future__ import annotations
from typing import Literal, Optional
from pydantic import BaseModel, Field

AircraftType = Literal["B737-800", "A320", "E175", "B757-200"]
FlightStatus = Literal["scheduled", "delayed", "cancelled", "diverted", "airborne", "landed"]


class Airport(BaseModel):
    icao: str
    name: str
    city: str
    lat: float
    lon: float
    hourly_capacity: int
    elevation_ft: int = 0


class Aircraft(BaseModel):
    tail_number: str
    aircraft_type: AircraftType
    seat_capacity: int
    base_airport: str
    min_turn_minutes: int = 45
    current_airport: Optional[str] = None


class ScheduledFlight(BaseModel):
    id: str
    flight_number: str
    origin: str
    destination: str
    scheduled_departure: str        # ISO-8601
    scheduled_arrival: str          # ISO-8601
    tail_number: str
    aircraft_type: AircraftType
    passengers: int = 120
    status: Optional[FlightStatus] = None


class CrewPairing(BaseModel):
    id: str
    captain_id: str
    first_officer_id: str
    base: str
    flights: list[str]
    report_time: str                # ISO-8601
    release_time: str               # ISO-8601
    duty_minutes: int
    flight_time_minutes: int
